system-tests/tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/local/TracingIntegrationTests.java

```diff
@@ -0,0 +1,133 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.system.tests.local;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.proto.trace.v1.Span;
+import org.eclipse.dataspaceconnector.common.annotations.OpenTelemetryIntegrationTest;
+import org.eclipse.dataspaceconnector.opentelemetry.OtlpGrpcServer;
+import org.eclipse.dataspaceconnector.system.tests.utils.FileTransferSimulationUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.eclipse.dataspaceconnector.system.tests.utils.GatlingUtils.runGatling;
+
+/**
+ * The role of this class is to test the opentelemetry traces.
+ * It only works if the opentelemetry java agent is attached.
+ * The default trace exporter used by the java agent is based is the OTLP exporter based on gRPC protocol.
+ * That's why an OtlpGrpcServer is used to collect the traces.
+ */
+@OpenTelemetryIntegrationTest
+public class TracingIntegrationTests extends FileTransferEdcRuntime {
+
+    static OtlpGrpcServer grpcServer;
+
+    String[] contractNegotiationSpanNames = new String[] {
```

Comment:
```suggestion
   List<String> contractNegotiationSpanNames = List.of(
```
nit, might be a bit nicer looking

Reply:
I will leave it like this as this is equivalent in my opinion.